src/types/mod.rs

```diff
@@ -119,10 +110,63 @@ impl System {
     }
 
     pub const fn get_all() -> [System; 4] {
-        [System::CH, System::AT, System::IT, System::SLO ]
+        [System::CH, System::AT, System::IT, System::SLO]
     }
+
+    // returns neighboring countries for a given system
+    pub fn adjacent(&self) -> Vec<System> {
+        match self {
+            System::CH => vec![System::AT, System::IT],
+            System::AT => vec![System::CH, System::IT, System::SLO],
+            System::IT => vec![System::AT, System::CH, System::SLO],
+            System::SLO => vec![System::AT, System::IT],
+        }
+    }
+    // returns an array of matching neighboring countries for two systems
+    pub fn shared_adjacency(&self, dest: System) -> Vec<System> {
+        let mut shared_neighbors: Vec<System> = Vec::new();
+        let origin: Vec<System> = self.adjacent();
+        let dest: Vec<System> = dest.adjacent();
+        for elem in origin {
+            if dest.contains(&elem) {
+                shared_neighbors.push(elem)
+            }
+        }
+        return shared_neighbors;
+    }
+
+    pub fn ajdacent_paths(&self, dest: System) -> Vec<Adjacency> {
+        let indirect = &mut self
+            .adjacent()
+            .into_iter()
+            .filter(|s| dest.adjacent().contains(s))
+            .map(|s| Adjacency::Indirect(*self, s, dest))
+            .collect::<Vec<Adjacency>>();
+
+        match &self.adjacent().contains(&dest) {
+            true => {
+                indirect.push(Adjacency::Direct(*self, dest));
+                return indirect.to_vec();
+            }
+            false => indirect.to_vec(),
+        }
+    }
+}
+
+//Values belonging to the systems. Requestor reference, a key for authentication, an url and an ID.
+#[derive(Debug, PartialEq, Serialize)]
+pub struct SystemConfig {
+    pub req_ref: &'static str,
+    pub key: &'static str,
+    pub url: &'static str,
+    pub id: System,
 }
 
+#[derive(Clone)]
```

Comment:
This allows us later on to match on the type of path, direct/indirect, and use a different algorithm according to this